vm-setup/roles/v1a2_integration_test/templates/controlplane_ubuntu.yaml

```diff
@@ -0,0 +1,101 @@
+apiVersion: cluster.x-k8s.io/v1alpha2
+kind: Machine
+metadata:
+  name: {{ CLUSTER_NAME }}-controlplane-0
+  labels:
+    cluster.x-k8s.io/control-plane: "true"
+    cluster.x-k8s.io/cluster-name: "{{ CLUSTER_NAME }}"
+spec:
+  version: {{ KUBERNETES_VERSION }}
+  bootstrap:
+    configRef:
+      apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+      kind: KubeadmConfig
+      name: {{ CLUSTER_NAME }}-controlplane-0
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+    kind: BareMetalMachine
+    name: {{ CLUSTER_NAME }}-controlplane-0
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+kind: BareMetalMachine
+metadata:
+  name: {{ CLUSTER_NAME }}-controlplane-0
+spec:
+  image:
+    url: {{ IMAGE_URL }}
+    checksum: {{ IMAGE_CHECKSUM }}
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+kind: KubeadmConfig
+metadata:
+  name: {{ CLUSTER_NAME }}-controlplane-0
+spec:
+  initConfiguration:
+    nodeRegistration:
+      name: {{ " '{{ ds.meta_data.name }}' " }}
+      kubeletExtraArgs:
+        node-labels: 'metal3.io/uuid={{ '{{ ds.meta_data.uuid }}' }}'
+  preKubeadmCommands:
+    - ip link set dev enp2s0 up
+    - dhclient enp2s0
+    - mv /tmp/akeys /home/ubuntu/.ssh/authorized_keys
+    - chown ubuntu:ubuntu /home/ubuntu/.ssh/authorized_keys
+    - apt update -y
+    - apt install net-tools -y
+    - apt install -y gcc linux-headers-$(uname -r)
```

Comment:
same thing for apt install, let's merge them all, I guess you'll just need to install curl first to download the repo keys, but the rest can probably be just merged

Reply:
Yes, merging them